contracts/multisig/src/key.rs

```diff
@@ -211,37 +215,40 @@ impl KeyDeserialize for KeyType {
     }
 }
 
-const ECDSA_COMPRESSED_PUBKEY_LEN: usize = 33;
-const ECDSA_UNCOMPRESSED_PUBKEY_LEN: usize = 65;
-
-const ED25519_PUBKEY_LEN: usize = 32;
-const ED25519_SIGNATURE_LEN: usize = 64;
+// It is possible for the same public key to be represented in different formats. For example
+// an ECDSA public key can be represented as compressed or uncompressed. This function
+// validates and normalizes the public key to a single format.
+fn validate_and_normalize_public_key(
+    key_type: KeyType,
+    pub_key: HexBinary,
+) -> Result<HexBinary, ContractError> {
+    match key_type {
+        KeyType::Ecdsa => Ok(k256::PublicKey::from_sec1_bytes(pub_key.as_slice())
+            .map_err(|_| ContractError::InvalidPublicKey)?
+            .to_sec1_bytes()
```

Comment:
Can you confirm if this using the compressed representation? tofn does [this](https://github.com/axelarnetwork/tofn/blob/main/src/ecdsa/mod.rs#L47) instead

Reply:
yes I tested and doesn't matter the input it throws the compressed representation. Internally it seems to be getting the bytes from the encoded point as well:

```
pub fn to_sec1_bytes(&self) -> Box<[u8]>
    where
        C: PointCompression,
        AffinePoint<C>: FromEncodedPoint<C> + ToEncodedPoint<C>,
        FieldBytesSize<C>: ModulusSize,
    {
        EncodedPoint::<C>::from(self).to_bytes()
    }
```

Reply:
Cool